### PR TITLE
Remove Bootstrap 2.3.1 Dependency

### LINKED ISSUE #178: Drop Bootstrap 2.3.1 Dependency
It looks like we're still using Bootstrap 2.3.1 in these views:

- `app/views/gradebooks/view.html.erb`
- `app/views/gradebooks/statistics.html.erb`
- `app/views/gradebooks/student.html.erb`
- `app/views/assessment_user_data/edit.html.erb`

As far as I can tell, the only reason why this has been kept around is for tooltips, which Bootstrap 3 also supports.

- [x] Replace Bootstrap 2.3.1 with Bootstrap 3.0.0 analogues everywhere

### LINKED ISSUE #320: Gradebook assessment names are cut off
When viewing the student gradebook, the column of assessment names is too narrow to fit even the word "Homework 0".

- [x] Fixable by making the viewport larger (it's currently capped at 970px)

### PR DESCRIPTION
**Pull request by jez**
Fixes #178. Fixes #320.

There were three views where Bootstrap 2.3.1 was being used:

- Assessment User Data (Edit)
    - Files affected:
        - app/views/assessment_user_data/edit.html.erb
    - This view was easy (it has very little complexity).
- Student Gradebook
    - Files affected:
        - app/assets/stylesheets/student_gradebook.css
        - app/assets/stylesheets/student_gradebook.css.scss
        - app/views/gradebooks/student.html.erb
    - Bootstrap 2.3.1 was being used both to style the layout of the
      page as well as the draw tooltips.
    - I updated the layout HTML to Bootstrap 3.0.0
    - I updated the tooltip usage to Bootstrap 3.0.0
    - I rewrote the student_gradebook.css file in SCSS
- Instructor Gradebook
    - Files affected:
        - app/views/layouts/application.html.erb
        - app/views/layouts/home.html.erb
        - app/views/gradebooks/view.html.erb
    - This page is using SlickGrid to display the instructor gradebook
      and was using jQuery UI + Bootstrap 2.3.1 Tooltip to display the
      tooltips.
    - So here's the dealio:
        - SlickGrid requires jQuery UI to sort columns
        - jQuery UI has features that enable tooltips
        - Bootstrap 2.3.1 integrates with jQuery UI tooltips
        - Bootstrap 3.0.0 has it's own tooltips implementation
        - jQuery UI interferes with Bootstrap 3.0.0 when trying to draw tooltips
    - I fixed this by adding no conflict method to the jQuery UI tooltip
      widget.
    - See here for more information
        - <http://stackoverflow.com/questions/13731400/jqueryui-tooltips-are-competing-with-twitter-bootstrap>

### CODE REVIEW

**Comment by jez** [comment]
Sorry for it being so many changes :/

File: app/views/gradebooks/view.html.erb [hunk 1]
```diff
@@ -58,9 +58,18 @@
...
-<% content_for :javascripts do %>
+
+<% content_for :js_widget_patch do %>
   <%= external_javascript_include_tag "jquery-ui", "1.10.2" %>
-  <%= external_javascript_include_tag "bootstrap-tooltip", "2.3.1" %>
+  <%# jQuery UI is required for SlickGrid, but Bootstrap provides a tooltip plugin %>
+  <%# The following must come after jQuery UI but before Bootstrap 3 is loaded %>
+  <%# to resolve namespace issues %>
+  <script>
+    $.widget.bridge('uitooltip', $.ui.tooltip);
```
Comment (icanberk) [inline app/views/gradebooks/view.html.erb]: can't we use bootstrap tooltips instead of jquery-ui? any idea what else is jquery-ui doing here?

Comment (dlbucci) [inline app/views/gradebooks/view.html.erb]: jquery-ui was being used in annotations until recently.  I don't know how many other things are using it.

**Comment by icanberk** [comment]
It's a very bulky (and useless imo) library. We should get rid of it if possible.

**Comment by dlbucci** [comment]
You never have to ask me to get rid of jquery code.

File: app/views/gradebooks/view.html.erb (hunk 1 above)
Comment (jez) [inline app/views/gradebooks/view.html.erb]: Did you guys even read the comment? It's required for SlickGrid.

**Comment by jez** [comment]
Also, I left a detailed description of why it's hard to remove jQuery UI in the PR description.

File: app/views/gradebooks/view.html.erb (hunk 1 above)
Comment (dlbucci) [inline app/views/gradebooks/view.html.erb]: I read Ilter's comment before reading yours. Sorry Jay-Z.

**Comment by jez** [comment]
I'm just gonna close this and start over :(

**Comment by dlbucci** [comment]
Why?

**Pull request closed without merging**